packages/framework/tinylicious-client/src/test/TinyliciousClient.spec.ts

```diff
@@ -277,4 +283,112 @@ describe("TinyliciousClient", () => {
         await corruptedAliasOp(runtimeOf(do1), "alias");
         assert(await dataCorruption);
     });
+
+    /**
+     * Scenario: Test if TinyliciousClient starts the container in write mode. TinyliciousClient will attempt
+     * to start the connection in write mode, but if the write permission is not available or the file is read
+     * only, mode will be read.
+     *
+     * Expected behavior: TinyliciousClientProps should start the container's with the connectionMode in `write`
+     */
+    it("can create a container in write mode", async () => {
+        const { container } = await tinyliciousClient.createContainer(schema);
+        const containerId = await container.attach();
+        await timeoutPromise((resolve) => container.once("connected", resolve), {
+            durationMs: 1000,
+            errorMsg: "container connect() timeout",
+        });
+
+        const { container: containerGet } = await tinyliciousClient.getContainer(
+            containerId,
+            schema
+        );
+
+        assert.strictEqual(
+            connectionModeOf(container),
+            "write",
+            "Creating a container is not in write mode"
+        );
+        assert.strictEqual(
+            connectionModeOf(containerGet),
+            "write",
+            "Getting a container is not in write mode"
+        );
+    });
+
+    /**
+     * Scenario: Test if TinyliciousClient with only read permission starts the container in read mode.
+     * TinyliciousClient will attempt to start the connection in write mode, but if the write permission
+     * is not available or the file is read only, mode will be read.
```

Comment:
Same as above: In your description can you be specific for this particular use case what plays out.